src/object/cli_ec.c

```diff
@@ -1316,6 +1321,78 @@ obj_ec_get_degrade(struct obj_reasb_req *reasb_req, uint16_t fail_tgt_idx,
 	return 0;
 }
 
+int
+obj_ec_singv_split(daos_obj_id_t oid, uint32_t shard, daos_size_t iod_size,
+		   struct daos_oclass_attr *oca, d_sg_list_t *sgl)
+{
+	uint64_t c_bytes = obj_ec_singv_cell_bytes(iod_size, oca);
+	uint32_t shard_idx = shard % obj_ec_data_tgt_nr(oca);
+	char	*data = sgl->sg_iovs[0].iov_buf;
+
+	if (shard_idx > 0)
+		memmove(data, data + shard_idx *c_bytes, c_bytes);
```

Comment:
Hmm, yes. ok, I thought about adding assertion here, but somehow need add IOD into the parameters. since this is a very short function, probably it is caller's reponsiblity to make sure this.

Though check iod_size might be ok here. Hmm, it is a bit strange to set iod_size = 0 for single value.  

I will think about it in the following patch.